### PR TITLE
samples: nrf9160: modem_shell: add command for heap usage statistics

### PR DESCRIPTION
**Pull request by tokangas**
Added a new command called `heap`, which prints out kernel and system heap usage statistics.

### CODE REVIEW

**Review by peknis — APPROVED** [review]

**Comment by tokangas** [comment]
I modified the code to use `mallinfo()` function instead of `malloc_stats()` and added printing of maximum system heap size. Also added the `heap` command to the documentation, ping @peknis.

The current output is like this:
```
mosh:~$ heap
kernel heap statistics:
free:             7804
allocated:         272
max. allocated:   1056

system heap statistics:
max. size:       81400
size:              248
free:              160
allocated:          88
```

**Comment by peknis** [comment]
> I modified the code to use `mallinfo()` function instead of `malloc_stats()` and added printing of maximum system heap size. Also added the `heap` command to the documentation, ping @peknis.

Still looks good, I thought you would add the command.

File: samples/nrf9160/modem_shell/README.rst [hunk 1]
```diff
@@ -714,6 +714,27 @@ Disable UARTs for power measurement purposes.
 
      uart during_sleep disable
 
+----
+
+Heap usage statistics
+=====================
+
+You can use the the ``heap`` command to print kernel and system heap usage statistics.
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/README.rst:722]:
```suggestion
You can use the ``heap`` command to print kernel and system heap usage statistics.
```

**Review by trantanen — APPROVED** [review]

**Comment by tokangas** [comment]
@rlubos I guess the kernel heap part would fit also in Zephyr kernel shell, what do you think?

**Comment by rlubos** [comment]
> @rlubos I guess the kernel heap part would fit also in Zephyr kernel shell, what do you think?

Yes I think it could be accepted upstream, although it's not really my area so can't tell for sure. Seems pretty useful though.

**Comment by tokangas** [comment]
> Yes I think it could be accepted upstream, although it's not really my area so can't tell for sure. Seems pretty useful though.

Ok, I think I'll create a PR for Zephyr. We can merge this anyway and later remove the kernel heap part if we get that into Zephyr.

**Pull request merged**